test/unit/pack_util_test/device/pack_util.hpp

```diff
@@ -66,6 +66,314 @@ namespace rocwmma
         return err;
     }
 
+    template <typename DataT, uint32_t VecSize>
+    ROCWMMA_DEVICE static inline bool padTest()
+    {
+        /**
+         * 8 bits [0x01, 0x02, 0x03, 0x04] -> 32 bits [0x00000001, 0x00000002, 0x00000003, 0x00000004]
+         * 16 bits [0x0001, 0x0002, 0x0003, 0x0004] -> 32 bits [0x00000001, 0x00000002, 0x00000003, 0x00000004]
+         * 32 bits, return the argument itself
+         * 64 bits, return the argument itself
+         **/
+        using PackUtil = rocwmma::PackUtil<DataT>;
+
+        bool err = false;
+        auto res = generateSeqVec<DataT, VecSize>();
+
+        if constexpr(PackUtil::Traits::PackRatio == 1)
+        {
+            err |= res != PackUtil::pad(res);
+        }
+        else
+        {
+            using PackedT   = typename PackUtil::Traits::PackedT;
+            using UnpackedT = typename PackUtil::Traits::UnpackedT;
+
+            auto paddedData = PackUtil::pad(res);
+            err |= !std::is_same_v<decltype(paddedData), VecT<PackedT, VecSize>>;
+            for(uint32_t i = 0; i < VecSize; i++)
+            {
+                auto value = get(paddedData, i);
+
+                // unpadded data is at idx 0 in the padded data by default
+                err |= *reinterpret_cast<DataT*>(&value) != static_cast<DataT>((float)i);
```

Comment:
Is the float cast to get around the integer ctor of bfloat16?

Reply:
yeah, building was failed since uint32_t cannot be cast to bf16.